crates/primitives/src/alloy_compat.rs

```diff
@@ -227,7 +233,31 @@ impl TryFrom<alloy_rpc_types::Transaction> for Transaction {
                 }))
             }
             #[cfg(feature = "optimism")]
-            Some(TxType::Deposit) => todo!(),
+            Some(TxType::Deposit) => {
+                Ok(Self::Deposit(TxDeposit {
+                    // TODO: Uses the wrong error here, need alloy ConversionError enum updated with new errors for this.
+                    source_hash: match &tx.other["sourceHash"] {
+                        serde_json::Value::String(source_hash) => source_hash.parse().map_err(|_| ConversionError::MissingBlockNumber)?,
+                        _ => return Err(ConversionError::MissingBlockNumber),
+                    },
+                    from: tx.from,
+                    to: match tx.to {
+                        Some(to) => TxKind::Call(to),
+                        None => TxKind::Create,
+                    },
+                    mint: match &tx.other["mint"] {
+                        serde_json::Value::String(mint) => mint.parse::<u128>().ok().and_then(|num| if num == 0 { None } else { Some(num) }),
```

Comment:
this will fail because mint is `U128`

Reply:
is there a convenient way to convert U128 to u128?

Reply:
are you sure mint is `U128`?

i see that is here: https://github.com/alloy-rs/alloy/blob/ed8a6baa2b1c404030a388633e9b7ee47825ed52/crates/rpc-types-eth/src/transaction/optimism.rs#L16

but i'm not sure if that struct is actually used. when i read the alloy block transactions, it's all strings and doesn't match the fields of that struct:
```
other: OtherFields {"depositReceiptVersion": String("0x1"), "mint": String("0x0"), "sourceHash": String("0x3a20754de0b2ef572fac9ca91ec3b182e61101c66d666897e8539e8b57b9a40c")}
```
i tried writing a test assuming it's a string and it works fine, see latest commit.